DDP-4875: Data request button

marat-mamytov opened the pull request:
Current PR is for frontend part of Data Request feature.

marat-mamytov committed:
4875 | move features into toolkit

charlottedye commented on ddp-workspace/projects/toolkit/src/lib/components/activity/new-activity-in-modal-button.component.ts:
```diff
@@ -0,0 +1,38 @@
+  launchActivity(): void {
+    this.dialog.open(NewActivityInModalComponent, {
+      width: '740px',
+      data: {activityGuid: this.activityGuid, createActivityInstance: true, modalName: this.modalName},
```
You could make this more flexible by calling this ActivityModalButtonComponent and having createActivityInstance be passed in the way activityGuid is.  I also don't think you need modalName here

charlottedye replied:
Instead of modalName, you can have a boolean for whether there should be a confirmation step at the end (maybe showFinalConfirmation) and then you could have confirmationMessageName (which would be Toolkit.DataRequest.Okay for our use case) and completeMessageName (which would be Toolkit.DataRequest.SubmitRequest for our use case).  See DataRequestComponent for more on how this would work...

charlottedye commented on ddp-workspace/projects/toolkit/src/lib/components/activity/data-request-activity.component.ts:
```diff
@@ -0,0 +1,70 @@
+    </div>`,
+  providers: [SubmitAnnouncementService, SubmissionManager]
+})
+export class DataRequestComponent extends ActivityComponent implements OnInit, OnDestroy, AfterViewInit {
```
I find the way things are split between DataRequestComponent and NewActivityInModalComponent kind of confusing.  I think you should be able to make this one component: ModalActivityComponent.  You'd use the modified version of the data object I described in NewActivityInModalComponent.  If showFinalConfirmation is true, you'd use the logic you have here for which button label to display, although you'd use confirmationMessageName instead of Toolkit.DataRequest.Okay and completeMessageName instead of Toolkit.DataRequest.SubmitRequest.  If showFinalConfirmation is false, you would just show the submit request button (in fact, you could probably improve this further by letting users specify the text for next, previous, submit, and confirm buttons.  You could assume that the confirmation step is always the final step).

charlottedye commented on ddp-workspace/projects/toolkit/src/lib/components/activity/data-request-activity.component.ts:
```diff
@@ -0,0 +1,70 @@
+    super(windowRef, renderer, submitService, analytics, document, injector);
+  }
+
+  public submitDataRequest(): void {
```
You'd just call this something like nextPage(), because that's really what it does.

charlottedye commented on ddp-workspace/projects/toolkit/src/lib/components/activity/data-request-activity.component.ts:
```diff
@@ -0,0 +1,70 @@
+    <div class="CenterChild" *ngIf="!isLoaded">
+      <mat-spinner></mat-spinner>
+    </div>
+    <div class="DataRequestModal--header" *ngIf="model && isLoaded">
```
See below--you could rename this ModalActivity--header or something

charlottedye commented on ddp-workspace/projects/toolkit/src/lib/components/activity/data-request-activity.component.ts:
```diff
@@ -0,0 +1,70 @@
+import { ActivityComponent, WindowRef, SubmitAnnouncementService, AnalyticsEventsService, SubmissionManager } from 'ddp-sdk';
+
+@Component({
+  selector: 'toolkit-data-request-activity',
```
See below--you could change this to toolkit-modal-activity

charlottedye commented on ddp-workspace/projects/toolkit/src/lib/components/activity/data-request-activity.component.ts:
```diff
@@ -0,0 +1,70 @@
+        </ng-container>
+      </div>
+      <div mat-dialog-actions align="end" *ngIf="model && isLoaded">
+        <button (click)="isLastStep ? closeModal() : submitDataRequest()"
```
See below: you'd change submitDataRequest to nextPage()

charlottedye commented on ddp-workspace/projects/toolkit/src/lib/components/activity/data-request-activity.component.ts:
```diff
@@ -0,0 +1,70 @@
+      </div>
+      <div mat-dialog-actions align="end" *ngIf="model && isLoaded">
+        <button (click)="isLastStep ? closeModal() : submitDataRequest()"
+                class="Button DataRequestModal--submit col-lg-6 col-md-6 col-sm-6 col-xs-12"
```
See below: you'd change DataRequestModal--submit to ModalActivity--submit

charlottedye commented on ddp-workspace/projects/toolkit/src/lib/components/activity/data-request-activity.component.ts:
```diff
@@ -0,0 +1,70 @@
+        <button (click)="isLastStep ? closeModal() : submitDataRequest()"
+                class="Button DataRequestModal--submit col-lg-6 col-md-6 col-sm-6 col-xs-12"
+                [innerText]="isLastStep
+                ? ('Toolkit.DataRequest.Okay' | translate)
```
See below: instead, you'd do this: `? (data.confirmationMessageName | translate)`

charlottedye commented on ddp-workspace/projects/toolkit/src/lib/components/activity/data-request-activity.component.ts:
```diff
@@ -0,0 +1,70 @@
+                class="Button DataRequestModal--submit col-lg-6 col-md-6 col-sm-6 col-xs-12"
+                [innerText]="isLastStep
+                ? ('Toolkit.DataRequest.Okay' | translate)
+                : ('Toolkit.DataRequest.SubmitRequest' | translate)"></button>
```
See below: instead, you'd do this: `: (data.completeMessageName | translate)"></button>`

charlottedye commented on ddp-workspace/projects/toolkit/src/lib/components/activity/new-activity-in-modal.component.ts:
```diff
@@ -0,0 +1,68 @@
+  private ngUnsubscribe = new Subject();
+
+  constructor(
+    public dialogRef: MatDialogRef<NewActivityInModalComponent>,
```
You could move anything relevant here into the ModalActivityComponent

charlottedye commented on ddp-workspace/projects/toolkit/src/lib/components/privacy-policy/prionPrivacyPolicy.component.ts:
```diff
@@ -0,0 +1,243 @@
+@Component({
+  selector: 'prion-privacy-policy',
+  template: `
+    <section class="PageContent-section">
```
For the record, if I had more time, I'd go back and make this privacy policy into an activity, but I hadn't thought of that at the time and I don't think we have time now.

charlottedye reviewed (changes requested):
See the comments--I think you could clean this up even a bit more by consolidating some things and making some things more general to fit additional use cases.  This is a big improvement, though!

marat-mamytov committed:
4875 | review fixes

marat-mamytov committed:
4875 | review fixes

marat-mamytov commented:
> See the comments--I think you could clean this up even a bit more by consolidating some things and making some things more general to fit additional use cases. This is a big improvement, though!

What was changed:
- selector name from 'toolkit-data-request-activity' to 'toolkit-modal-activity' and made modal activity more general and configurable
- data object for modal: removed createActivityInstance and modalName props and added new props for showing the confirmation step and buttons
- modal activity methods
- classes from DataRequestModal to ModalActivity
- moved code creating new activity instance to the button click event -> this way I can pass this new instanceGuid instead of activityGuid into the modal
- combined all modal code into one component ModalActivityComponent
@charlottedye Please advise whether all above is ok.

marat-mamytov committed:
4875 | fix name

marat-mamytov committed:
4875 | fix styles after components relocation to the toolkit

marat-mamytov committed:
4875 | create activity insance onclick and pass instance guid to modal

marat-mamytov committed:
4875 | combine modal code in one component

marat-mamytov committed:
4875 |return error handler when instance is not created

marat-mamytov committed:
4875 | remove extra import

marat-mamytov committed:
4875 | remove extra import

charlottedye committed:
Prion: DDP-4875: Fixed typo in CSS file
